zerver/models.py

```diff
@@ -1091,6 +1091,13 @@ def can_admin_user(self, target_user: 'UserProfile') -> bool:
     def __str__(self) -> str:
         return "<UserProfile: %s %s>" % (self.email, self.realm)
 
+    @property
+    def is_new_member(self) -> bool:
+        diff = (timezone_now() - self.date_joined).days
+        if diff < self.realm.waiting_period_threshold:
+            return True
+        return False
```

Comment:
Can you extract this refactor as its own preparatory commit at the start of the branch?  That way we can merge it early now while we work through the rest of this.